pyuvdata/uvcal.py

```diff
@@ -231,6 +233,202 @@ def set_unknown_cal_type(self):
+                history_update_string += 'times'
+            n_selects += 1
+
+            time_inds = np.zeros(0, dtype=np.int)
+            for jd in times:
+                if jd in self.time_array:
+                    time_inds = np.append(time_inds, np.where(self.time_array == jd)[0])
+                else:
+                    raise ValueError('Time {t} is not present in the time_array'.format(t=jd))
+
+            time_inds = list(sorted(set(list(time_inds))))
+            self.Ntimes = len(time_inds)
+            self.time_array = self.time_array[time_inds]
+            # should we adjust the time_range parameter?
+
+            time_separation = self.time_array[1:] - self.time_array[:-1]
+            if np.min(time_separation) < np.max(time_separation):
+                warnings.warn('Selected times are not evenly spaced. This '
+                              'will make it impossible to write this data out to '
+                              'some file types')
+
+            self.flag_array = self.flag_array[:, :, :, time_inds, :]
+            if self.cal_type == 'delay':
+                self.quality_array = self.quality_array[:, :, time_inds, :]
+                self.delay_array = self.delay_array[:, :, time_inds, :]
+            else:
+                self.quality_array = self.quality_array[:, :, :, time_inds, :]
+                self.gain_array = self.gain_array[:, :, :, time_inds, :]
+
+            if self.input_flag_array is not None:
+                self.input_flag_array = self.input_flag_array[:, :, :, time_inds, :]
+
+        if freq_chans is not None:
+            freq_chans = uvutils.get_iterable(freq_chans)
+            if frequencies is None:
+                frequencies = self.freq_array[0, freq_chans]
+            else:
+                frequencies = uvutils.get_iterable(frequencies)
+                frequencies = np.sort(list(set(frequencies) |
+                                      set(self.freq_array[0, freq_chans])))
+
+        if frequencies is not None:
+            frequencies = uvutils.get_iterable(frequencies)
+            if n_selects > 0:
+                history_update_string += ', frequencies'
+            else:
+                history_update_string += 'frequencies'
+            n_selects += 1
+
+            freq_inds = np.zeros(0, dtype=np.int)
+            # this works because we only allow one SPW. This will have to be reworked when we support more.
+            freq_arr_use = self.freq_array[0, :]
+            for f in frequencies:
+                if f in freq_arr_use:
+                    freq_inds = np.append(freq_inds, np.where(freq_arr_use == f)[0])
+                else:
+                    raise ValueError('Frequency {f} is not present in the freq_array'.format(f=f))
+
+            freq_inds = list(sorted(set(list(freq_inds))))
+            self.Nfreqs = len(freq_inds)
+            self.freq_array = self.freq_array[:, freq_inds]
+            # should we adjust the freq_range parameter?
```

Comment:
Ok, I think I get the general idea. I'm not sure how difficult it would be to make it a delay-only parameter. It actually sounds quite difficult with how we have the object set up. I'm not sure it's harming anything if it's there for gain type, so I'll leave that up to you two.